instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -302,7 +311,8 @@ def _end_span_after_iterating(iterable, span, tracer, token):
         if close:
             close()
         span.end()
-        context.detach(token)
+        if token is not None:
```

Comment:
I think we should check for `token` not being None in every instrumentation. I see some check for `ctx` but then run operations on `token`.